src/Microsoft.Diagnostics.Monitoring.EventPipe/Counters/CounterPayload.cs

```diff
@@ -86,7 +86,38 @@ protected MeterPayload(DateTime timestamp,
         {
         }
 
-        public override bool IsMeter => true;
+        public sealed override bool IsMeter => true;
+    }
+
+    internal abstract class MeterInstrumentDeltaMeasurementPayload : MeterPayload
+    {
+        protected MeterInstrumentDeltaMeasurementPayload(
+            DateTime timestamp,
+            CounterMetadata counterMetadata,
+            string displayName,
+            string unit,
+            double value,
+            CounterType counterType,
+            string valueTags,
+            EventType eventType)
+            : base(timestamp, counterMetadata, displayName, unit, value, counterType, valueTags, eventType)
+        {
+        }
+
+        public abstract bool SupportsDelta { get; }
```

Comment:
This feels confusing that the type is named "MeterInstrumentDeltaMeasurementPayload" and yet it might have SupportsDelta return false. Does the "Delta" in the type name have a different meaning than the Delta in the property name?

Reply:
I moved it on to `MeterPayload` with a default of `false`. For instruments where delta is supported they now return `true`.